internal/events/aggregator.go

```diff
@@ -553,3 +591,44 @@ func (ag *aggregator) resolveBlobs(ctx context.Context, data fftypes.DataArray)
 	return true, nil
 
 }
+
+func (ag *aggregator) rewindForBlobArrival(ctx context.Context, blobHash *fftypes.Bytes32) error {
+
+	batchIDs := make(map[fftypes.UUID]bool)
+
+	// We need to work out what pins potentially are unblocked by the arrival of this data
+
+	// Find any data associated with this blob
```

Comment:
Do we need to consider caching these data/message lookups?

Reply:
Potentially, yes. But I'm going to defer this. Current assumption is that blobs are going to be used when there is heavy data, and the resulting saving in caching for these small data object lookups is hence less significant.